tests/test_hamiltonians.py

```diff
@@ -69,10 +69,10 @@ def transformation_d(a, b, use_eye=False):
         H2 = hamiltonians.XXZ(nqubits=2, delta=1, backend=backend)
         mH1, mH2 = backend.to_numpy(H1.matrix), backend.to_numpy(H2.matrix)
     else:
-        if backend.platform == "tensorflow":
-            pytest.skip("Tensorflow does not support operations with sparse matrices.")
-        elif backend.name == "pytorch":
-            pytest.skip("Pytorch does not support operations with sparse matrices.")
+        if backend.platform in ["tensorflow", "pytorch"]:
+            pytest.skip(
+                "Tensorflow and Pytorch do not support operations with sparse matrices."
+            )
```

Comment:
This is not really for this PR, but I think that both of them support sparse matrices to some extent [https://pytorch.org/docs/stable/generated/torch.sparse.mm.html](https://pytorch.org/docs/stable/generated/torch.sparse.mm.html)
[https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/sparse-mat-mul](https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/sparse-mat-mul)

I am not sure what was causing problems here, but I guess it is more a matter of how the operation is called rather than the impossibility to perform it. Anyway, this is probably out of the scope of this PR.

Reply:
I just opened an issue with this comment.